SEDaily-IOS/API.swift

```diff
@@ -154,186 +144,136 @@ extension API {
-                        case API.Types.new:
-                            item.isNew = true
-                        default:
-                            break
-                        }
-                        item.save()
-                    }
-                    else {
-                        // Just update the existing item
-                        let recommended = existingItem!.isRecommended
-                        
-                        existingItem?.updateFrom(item: item)
-                        
-                        existingItem?.update(isRecommended: recommended)
-                        
-                        switch type {
-                        case API.Types.top:
-                            existingItem?.update(isTop: true)
-                        case API.Types.new:
-                            existingItem?.update(isNew: true)
-                        default:
-                            break
-                        }
+                
+                var data: [podcastModel] = []
+                let this = JSON(responseData)
+                for (_, subJson):(String, JSON) in this {
+                    guard let jsonData = try? subJson.rawData() else { continue }
+                    let newObject = try? JSONDecoder().decode(podcastModel.self, from: jsonData)
+                    if let newObject = newObject {
+                        data.append(newObject)
                     }
                 }
-                completion(hasChanges)
+                onSucces(data)
             case .failure(let error):
-                log.error(error)
+                log.error(error.localizedDescription)
                 Tracker.logGeneralError(error: error)
-                completion(false)
+                onFailure(.GeneralFailure)
             }
         }
     }
-    
-    func getRecommendedPosts(createdAtBefore beforeDate: String = "", completion: @escaping (_ hasChanges: Bool) -> Void) {
-        let urlString = rootURL + Endpoints.recommendations
+}
+
+// MARK: - MVVM Getters
+extension API {
+    func getPosts(type: String = "",
+                  createdAtBefore beforeDate: String = "",
+                  tags: String = "-1",
+                  categories: String = "",
+                  onSucces: @escaping ([Podcast]) -> Void,
+                  onFailure: @escaping (APIError?) -> Void) {
+        var type = type
         
-        let user = User.getActiveUser()
-        guard let userToken = user.token else { return }
+        let user = UserManager.sharedInstance.getActiveUser()
+        let userToken = user.token
         let _headers : HTTPHeaders = [
             Headers.authorization:Headers.bearer + userToken,
-        ]
+            ]
 
-        typealias model = PodcastModel
-        
-        // Variable to check if this function returns changes
-        var hasChanges = false
+        if userToken == "" && type == PodcastTypes.recommended.rawValue {
```

Comment:
Consider using the string method .isEmpty rather than using != "" or == "".

Reply:
Good catch!